pkg/public/strategies.go

```diff
@@ -0,0 +1,209 @@
+
+	// create or update aws strategies config map
+	awsStratConfig := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cloud-resources-aws-strategies",
+			Namespace: namespace,
+		},
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, client, awsStratConfig, func() error {
+		// ensure data is not nil, create a default config map
+		if awsStratConfig.Data == nil {
+			// default config map contains a `development` and `production` tier
+			awsStratConfig.Data = buildDefaultAWSConfigMap()
+		}
+
+		// check to ensure postgres and redis key is included in config data
+		// build default postgres and redis key, contains a `development` and `production` tier
+		if _, ok := awsStratConfig.Data["postgres"]; !ok {
+			awsStratConfig.Data["postgres"] = buildDefaultAWSStratValue()
+		}
+		if _, ok := awsStratConfig.Data["redis"]; !ok {
+			awsStratConfig.Data["redis"] = buildDefaultAWSStratValue()
+		}
+
+		// marshal strategies, updating existing strategies with new values
+		postgresStrategy, err := reconcilePostgresStrategy(awsStratConfig.Data["postgres"], tier, backupWindow, maintenanceWindow)
+		if err != nil {
+			return errorUtil.Wrapf(err, "failed to reconcile postgres strategy")
+		}
+		redisStrategy, err := reconcileRedisStrategy(awsStratConfig.Data["redis"], tier, backupWindow, maintenanceWindow)
+		if err != nil {
+			return errorUtil.Wrapf(err, "failed to reconcile redis strategy")
+		}
+
+		// setting postgres and redis values to be updated strategies
+		awsStratConfig.Data["postgres"] = postgresStrategy
+		awsStratConfig.Data["redis"] = redisStrategy
+
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to update aws strategy config map : %v", err)
+	}
+	return nil
+}
+
+// reconciles Postgres strategy
+// unmarshalls raw strategy to create db instance input
+// checks found values from current config map vs expected values
+// marshalls create db instance input back to raw strategy
+func reconcilePostgresStrategy(config, tier, backupWindow, maintenanceWindow string) (string, error) {
+	// unmarshall config data to type of cro aws strategy config
+	var rawStrategy map[string]*croAWS.StrategyConfig
+	if err := json.Unmarshal([]byte(config), &rawStrategy); err != nil {
+		return "", errorUtil.Wrapf(err, "failed to unmarshal strategy mapping for postgres resource")
+	}
+
+	// unmarshall the create strategy from the raw strategy tier to type create db instance input
+	rdsCreateConfig := &rds.CreateDBInstanceInput{}
+	if err := json.Unmarshal(rawStrategy[tier].CreateStrategy, rdsCreateConfig); err != nil {
+		return "", errorUtil.Wrapf(err, "failed to unmarshal aws rds cluster config")
+	}
+
+	// check create db instance input against expected values
+	// backup window and maintenance window are expected values via RHMIconfig
+	// todo we may want to handle more strategy input, we should add functionality to check every value
```

Comment:
We do similar checks in other areas in CRO, right? Is there any way we can re-use that logic?